app/models/concerns/weaknesses/scopes.rb

```diff
@@ -21,6 +21,26 @@ module Weaknesses::Scopes
       ).order(risk: :desc, state: :asc)
     }
     scope :latest, -> { where latest_id: nil }
+
+    def process_owner_parents
+      parents = []
+
+      process_owners.map do |po|
+        parents << po.parent.full_name if po.parent
+      end
+
+      parents
+    end
```

Comment:
```suggestion
    def process_owner_parents
      process_owners.map do |po|
        po.parent.full_name if po.parent
      end.compact
    end
```
Es un `map`, devuelve el array. Lo que estabas haciendo es para `each`, no me di cuenta antes.

Y otra cosa, no lo pongo en el suggestion para que no te rompa el resto, pero NO DEBERÍAS devolver el full_name, sino el usuario y después le hacés full_name donde te haga falta. Sino imaginate que en otro lugar necesitamos lo mismo, pero mostrando el email, tenemos que hacer otro método y así por cada caso.